firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/UpdateApkClient.java

```diff
@@ -47,57 +44,50 @@
   private TaskCompletionSource<File> downloadTaskCompletionSource;
   private CancellationTokenSource downloadCancellationTokenSource;
   private final Executor downloadExecutor;
-  private final Handler downloadHandler;
   private TaskCompletionSource<Void> installTaskCompletionSource;
   private final FirebaseApp firebaseApp;
-  private UpdateTaskImpl updateTask;
-  private TaskCompletionSource updateAppTaskCompletionSource;
 
   public UpdateApkClient(@NonNull FirebaseApp firebaseApp) {
     this.downloadExecutor = Executors.newSingleThreadExecutor();
-    this.downloadHandler = HandlerCompat.createAsync(Looper.getMainLooper());
     this.firebaseApp = firebaseApp;
   }
 
   public void updateApk(
-      @NonNull String downloadUrl,
-      @NonNull Activity currentActivity,
       @NonNull UpdateTaskImpl updateTask,
-      @NonNull TaskCompletionSource updateAppTaskCompletionSource) {
-    this.updateTask = updateTask;
-    this.updateAppTaskCompletionSource = updateAppTaskCompletionSource;
-    downloadApk(downloadUrl)
+      @NonNull String downloadUrl,
+      @NonNull Activity currentActivity) {
```

Comment:
Similar to the comment above, feels a bit weird that updateApk is taking in a task instead of returning a task.